nautobot/dcim/views.py

```diff
@@ -2365,16 +2374,23 @@ class ModuleUIViewSet(BulkComponentCreateUIViewSetMixin, NautobotUIViewSet):
     serializer_class = serializers.ModuleSerializer
     table_class = tables.ModuleTable
     component_model = None
-
-    def get_action(self):
-        if self.component_model:
-            method = self.request.method.lower()
-            if method == "get":
-                return "view"
-            else:
-                return "change"
-
-        return super().get_action()
+    custom_action_permission_map = {
+        "consoleports": "view",
+        "consoleserverports": "view",
+        "powerports": "view",
+        "poweroutlets": "view",
+        "interfaces": "view",
+        "frontports": "view",
+        "rearports": "view",
+        "modulebays": "view",
```

Comment:
Why no custom `get_required_permission`override for these to enforce e.g. `view_consoleport`, etc.?

Same question below actually, don't we need `add_consoleport` and similar permissions?